accepted/0000-workspace-run-commands.md

```diff
@@ -0,0 +1,67 @@
+- Start Date: 2017-09-21
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+Allow Yarn CLI to execute a package script on each packages from the workspace root.  
+
+# Motivation  
+
+"Lerna". Lerna does a great job for handling monorepos. Since Yarn has a built-in 
+workspaces feature, we could use some of the functionalities that being used in 
+lerna (not entirely though!) for handling the monorepo more efficiently.  
+
+Just like installing all dependencies from one place (workspace root), it'd be great 
+to execute all the sub-package scripts from the workspace root using a single command. 
+Not only this would make a less back-and-forth traveling between the packages to 
+execute scripts, but also it'll help a lot for CI/CD configuration. 
+
+
+# Detailed design  
+
+As @BYK mentioned [here](https://github.com/yarnpkg/yarn/issues/4467#issuecomment-330873337), 
+we could create two commands for this feature.  
+
+* `yarn workspaces list`  
+* `yarn workspaces run <command>`  
+* `yarn workspaces add <workspace_name>`  
+* `yarn workspace remove <workspace_name> --opts`
+
+By introducing `workpaces` command, it brings a modular approach from the CLI's perspective.  
+
+## `yarn workspaces list`  
+
+This command lists out the package names inside that workspace.  In the case of multiple workspace configuration, we'll need to pass the workspace name like `yarn workspaces <workspace_name> list`. By defaut, when no workspace name is not given, it'll pick the first workspace name from the root package.json.  
+
+## `yarn workspaces run <command>`  
+
+This command will execute the specified command in all the packages under the workspace. Before we run the scripts, it should make sure that all packages has the script specified in the package.json. Otherwise, it should throw an error specifying which pacakge is missing that script.  
+
+Just like the `list` command, by default, it'll select the first workspace in case of no workspace name is provided.  
+
+The important thing here to note that the ordering of the execution. It must be executed in the same dependency order. This will assure that we could use a single "build" that builds inter-dependant packages without any problem.  
+
+
+## `yarn workspaces add <workspace_name>`  
+
+This would create the folder if it doesn't exist and make sure it is included in the top-level `workspaces` config.  
+
+## `yarn workspace remove <workspace_name> --opts`  
+
+Remove a specific workspace from the config. And by passing `-D` as an option, it'll delete the folder.
```

Comment:
For simplicity I would focus only on list/run commands.

Add/delete are less utility and more like sugar on top of editing package.json and adding/removing folders.
Also I don't like inconsistency workspaces/workspace for those commands
And in Yarn we prefer using explicit flags and not shorts `--delete-folder` would be more consistent with other commands

Reply:
On a second thought, yeah, you're right. This doesn't gives any extra convenience. Removed!